src/kernels/MIOpenCol2Im2d.cl

```diff
@@ -33,51 +33,61 @@
-                       const int col_h,
-                       const int col_w,
-                       const int wei_h,
-                       const int wei_w,
-                       const int pad_h,
-                       const int pad_w,
-                       const int stride_h,
-                       const int stride_w,
-                       const int dilation_h,
-                       const int dilation_w,
-                       const int height,
-                       const int width,
-                       global _FLOAT* im,
-                       const int im_offset)
+#ifndef MIOPEN_USE_64BIT_INDEX
+#error "MIOPEN_USE_64BIT_INDEX must be defined"
+#endif
+
+#if MIOPEN_USE_64BIT_INDEX
+typedef ulong index_t;
+#else
+typedef uint index_t;
+#endif
+
+__kernel void Col2Im2dU(global _FLOAT* col,
+                        const uint col_h,
+                        const uint col_w,
+                        const uint wei_h,
+                        const uint wei_w,
+                        const uint pad_h,
+                        const uint pad_w,
+                        const uint stride_h,
+                        const uint stride_w,
+                        const uint dilation_h,
+                        const uint dilation_w,
+                        const uint height,
+                        const uint width,
+                        global _FLOAT* im,
+                        const uint im_offset)
 {
     global _FLOAT* im_off = im + im_offset;
-    int gid               = (int)get_global_id(0);
+    uint gid              = (uint)get_global_id(0);
 
-    int im_ch  = gid / (width * height);
-    int im_pix = gid % (width * height);
-    int im_h   = (im_pix / width) + pad_h;
-    int im_w   = (im_pix % width) + pad_w;
+    uint im_ch  = gid / (width * height);
+    uint im_pix = gid % (width * height);
+    uint im_h   = (im_pix / width) + pad_h;
+    uint im_w   = (im_pix % width) + pad_w;
 
-    int start_h = (im_h < dilation_h * (wei_h - 1) + 1)
-                      ? 0
-                      : (im_h - (dilation_h * (wei_h - 1) + 1)) / stride_h + 1;
-    int end_h   = min(col_h, im_h / stride_h + 1);
-    int start_w = (im_w < dilation_w * (wei_w - 1) + 1)
-                      ? 0
-                      : (im_w - (dilation_w * (wei_w - 1) + 1)) / stride_w + 1;
-    int end_w   = min(col_w, im_w / stride_w + 1);
+    uint start_h = (im_h < dilation_h * (wei_h - 1) + 1)
+                       ? 0
+                       : (im_h - (dilation_h * (wei_h - 1) + 1)) / stride_h + 1;
+    uint end_h   = min(col_h, im_h / stride_h + 1);
+    uint start_w = (im_w < dilation_w * (wei_w - 1) + 1)
+                       ? 0
+                       : (im_w - (dilation_w * (wei_w - 1) + 1)) / stride_w + 1;
+    uint end_w   = min(col_w, im_w / stride_w + 1);
 
-    int ch_offset = im_ch * col_w * col_h * wei_w * wei_h;
+    index_t ch_offset = (index_t)(im_ch * col_w * col_h) * (index_t)(wei_w * wei_h);
```

Comment:
This prevents overflow in Col2Im2D. `index_t` depends on value of `MIOPEN_USE_64BIT_INDEX`.